code/modules/vehicles/arc/verbs.dm

```diff
@@ -0,0 +1,119 @@
+/obj/vehicle/multitile/arc/proc/toggle_antenna(mob/toggler)
+	set name = "Toggle Sensor Antenna"
+	set desc = "Raises or lowers the external sensor antenna. While raised, the ARC cannot move."
+	set category = "Vehicle"
+
+	var/mob/user = toggler || usr
+	if(!user || !istype(user))
+		return
+
+	var/obj/vehicle/multitile/arc/vehicle = user.interactee
+	if(!istype(vehicle))
+		return
+
+	var/seat
+	for(var/vehicle_seat in vehicle.seats)
+		if(vehicle.seats[vehicle_seat] == user)
+			seat = vehicle_seat
+			break
+
+	if(!seat)
+		return
+
+	if(vehicle.health < initial(vehicle.health) * 0.5)
+		to_chat(user, SPAN_WARNING("[vehicle]'s hull is too damaged to operate!"))
+		return
+
+	var/obj/item/hardpoint/support/arc_antenna/antenna = locate() in vehicle.hardpoints
+	if(!antenna)
+		to_chat(user, SPAN_WARNING("[vehicle] has no antenna mounted!"))
+		return
+
+	if(antenna.deploying)
+		return
+
+	if(antenna.health <= 0)
+		to_chat(user, SPAN_WARNING("[antenna] is broken!"))
+		return
+
+	if(vehicle.antenna_deployed)
+		to_chat(user, SPAN_NOTICE("You begin to retract [antenna]..."))
+		antenna.deploying = TRUE
+		if(!do_after(user, max(vehicle.antenna_toggle_time - antenna.deploy_animation_time, 1 SECONDS), target = vehicle))
+			to_chat(user, SPAN_NOTICE("You stop retracting [antenna]."))
+			antenna.deploying = FALSE
+			return
+
+		antenna.retract_antenna()
+		addtimer(CALLBACK(vehicle, PROC_REF(finish_antenna_retract), vehicle, user), antenna.deploy_animation_time)
+
+	else
+		to_chat(user, SPAN_NOTICE("You begin to extend [antenna]..."))
+		antenna.deploying = TRUE
+		if(!do_after(user, max(vehicle.antenna_toggle_time - antenna.deploy_animation_time, 1 SECONDS), target = vehicle))
+			to_chat(user, SPAN_NOTICE("You stop extending [antenna]."))
+			antenna.deploying = FALSE
+			return
+
+		antenna.deploy_antenna()
+		addtimer(CALLBACK(vehicle, PROC_REF(finish_antenna_deploy), vehicle, user), antenna.deploy_animation_time)
+
+/obj/vehicle/multitile/arc/proc/finish_antenna_retract(obj/vehicle/multitile/arc/vehicle, mob/user)
```

Comment:
Does it need an argument that is probably `src` anyway?

Reply:
Probably not, no.